dags/nyc_taxi_dag.py

```diff
@@ -1,138 +1,65 @@
-
-A detailed tutorial is available at https://community.crate.io/t/cratedb-and-apache-airflow-building-a-data-ingestion-pipeline/926
-
-Prerequisites
--------------
-In the CrateDB schema "nyc_taxi", the tables "load_files_processed",
-"load_trips_staging" and "trips" need to be present before running the DAG.
-You can retrieve the CREATE TABLE statements from the file setup/taxi-schema.sql
-in this repository.
-"""
-import logging
 from pathlib import Path
-import pendulum
-from airflow import DAG
+from datetime import datetime
+from airflow.decorators import task, dag
 from airflow.providers.postgres.operators.postgres import PostgresOperator
-from airflow.providers.postgres.hooks.postgres import PostgresHook
-from airflow.providers.http.operators.http import SimpleHttpOperator
-from airflow.operators.python import PythonOperator
-
-
-def get_processed_files(_ti):
-    pg_hook = PostgresHook(postgres_conn_id="cratedb_demo_connection")
-    records = pg_hook.get_records(
-        sql="SELECT file_name FROM nyc_taxi.load_files_processed"
-    )
-
-    # flatten nested list as there is only one column
-    return list(map(lambda record: record[0], records))
-
-
-def clean_data_urls(ti):
-    data_urls_raw = ti.xcom_pull(task_ids="download_data_urls", key="return_value")
-
-    data_urls = data_urls_raw.split("\n")
-    # we only import Yellow tripdata for now due to different CSV schemas
-    data_urls_filtered = filter(lambda element: "yellow" in element, data_urls)
-
-    return list(data_urls_filtered)
-
-
-def identitfy_missing_urls(ti):
-    data_urls_processed = ti.xcom_pull(task_ids="get_processed_files")
-    data_urls_available = ti.xcom_pull(task_ids="clean_data_urls")
-
-    return list(set(data_urls_available) - set(data_urls_processed))
-
-
-def process_new_files(ti):
-    missing_urls = ti.xcom_pull(task_ids="identitfy_missing_urls")
-
-    for missing_url in missing_urls:
-        logging.info(missing_url)
-
-        file_name = missing_url.split("/").pop()
-
-        PostgresOperator(
-            task_id=f"copy_{file_name}",
-            postgres_conn_id="cratedb_demo_connection",
-            sql=f"""
-                    COPY nyc_taxi.load_trips_staging
-                    FROM '{missing_url}'
-                    WITH (format = 'csv', empty_string_as_null = true)
-                    RETURN SUMMARY;
-                """,
-        ).execute({})
-
+from airflow.operators.bash import BashOperator
+from airflow.models import Variable
+
+
+@dag(dag_id="nyc-taxi-parquet",
+schedule_interval='@monthly',
+start_date=datetime(2009, 1, 1),
+catchup=True)
+def taskflow():
+    @task(task_id='get_month_year')
+    def get_month_year(ds=None):
+        currentMonth = int(ds.split('-')[1]) - 2
+        currentYear = int(ds.split('-')[0])
+        if currentMonth == 12:
+            currentYear = currentYear - 1
+        if currentMonth < 10:
+            currentMonth = f'0{currentMonth}'
+
+        return f'_{currentYear}-{currentMonth}'
```

Comment:
This doesn't seem to work for all cases. For January, `currentMonth` would be `-2`?

Instead of handling the date subtraction and all edge cases manually, I suggest using a library like [pendulum](https://pendulum.eustace.io).
With that, it should be much easier:
```python
import pendulum
ds = '2022-01-01'
timestamp = pendulum.parse(ds)
timestamp = timestamp.subtract(months=1)
timestamp.format('_YYYY-MM')
```